vello_shaders/shader/fine.wgsl

```diff
@@ -987,18 +988,23 @@ fn main(
                 cmd_ix += 2u;
             }
             case CMD_BEGIN_CLIP: {
+                let is_non_isolated = (tag & NON_ISOLATED_BLENDING_FLAG) != 0;
+                var rgba_mul = vec4(0.0);
+                if is_non_isolated {
+                    rgba_mul = vec4(1.0);
+                } 
                 if clip_depth < BLEND_STACK_SPLIT {
                     for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
                         blend_stack[clip_depth][i] = pack4x8unorm(rgba[i]);
-                        rgba[i] = vec4(0.0);
-                    }
+                        rgba[i] *= rgba_mul;
+                    }                        
                 } else {
                     let blend_in_scratch = clip_depth - BLEND_STACK_SPLIT;
                     let local_tile_ix = local_id.x * PIXELS_PER_THREAD + local_id.y * TILE_WIDTH;
                     let local_blend_start = blend_offset + blend_in_scratch * TILE_WIDTH * TILE_HEIGHT + local_tile_ix;
                     for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
                         blend_spill[local_blend_start + i] = pack4x8unorm(rgba[i]);
-                        rgba[i] = vec4(0.0);
+                        rgba[i] *= rgba_mul;
```

Comment:
Obviously this is assuming that doing a floating point multiply is extremely cheap, right?
At least in the non-isolated case, not doing anything at all is of course more ideal. Although that would of course involve hoisting the condition, leading to code duplication.